File: GXpert/GXpert.Web/Modules/Institute/Department/DepartmentColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace GXpert.Institute.Columns;

[ColumnsScript("Institute.Department")]
[BasedOnRow(typeof(DepartmentRow), CheckNames = true)]
public class DepartmentColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    [EditLink]
    public string Title { get; set; }
    public string InstituteName { get; set; }
    public string HODName { get; set; }
    public string Description { get; set; }
   
}

File: GXpert/GXpert.Web/Modules/ServerTypes/Institute/DepartmentColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { DepartmentRow } from "./DepartmentRow";

export interface DepartmentColumns {
    Id: Column<DepartmentRow>;
    Title: Column<DepartmentRow>;
    InstituteName: Column<DepartmentRow>;
    HodName: Column<DepartmentRow>;
    Description: Column<DepartmentRow>;
}

export class DepartmentColumns extends ColumnsBase<DepartmentRow> {
    static readonly columnsKey = 'Institute.Department';
    static readonly Fields = fieldsProxy<DepartmentColumns>();
}